### PR TITLE
Docs: reword sentences about dbm submodule traits

### PR DESCRIPTION
**Pull request by erlend-aasland**
- don't repeatedly say that keys and values are coerced into bytes

----
📚 Documentation preview 📚: https://cpython-previews--114609.org.readthedocs.build/

### CODE REVIEW

**Comment by erlend-aasland** [comment]
Follow-up to address @serhiy-storchaka's concerns.

**Comment by erlend-aasland** [comment]
The general info is here, @serhiy-storchaka:

https://github.com/python/cpython/blob/699779256ec4d4b8afb8211de08ef1382c78c370/Doc/library/dbm.rst?plain=1#L89-L91

File: Doc/library/dbm.rst [hunk 1]
```diff
@@ -353,10 +347,10 @@ This module can be used with the "classic" NDBM interface or the
 
 The :mod:`dbm.dumb` module provides a persistent :class:`dict`-like
 interface which is written entirely in Python.
-Unlike other :mod:`dbm` backends, such as :mod:`dbm.gnu`, no
+Unlike other modules such as :mod:`dbm.gnu`, no
```
Comment (erlend-aasland) [inline Doc/library/dbm.rst]:
```suggestion
Unlike other :mod:`dbm` backends, such as :mod:`dbm.gnu`, no
```

File: Doc/library/dbm.rst [hunk 2]
```diff
@@ -355,8 +349,8 @@ The :mod:`dbm.dumb` module provides a persistent :class:`dict`-like
 interface which is written entirely in Python.
 Unlike other :mod:`dbm` backends, such as :mod:`dbm.gnu`, no
 external library is required.
-As with other :mod:`dbm` backends,
-the keys and values are always stored as :class:`bytes`.
+
+The module defines the following:
```
Comment (erlend-aasland) [inline Doc/library/dbm.rst]:
```suggestion
```

**Comment by erlend-aasland** [comment]
Are you ok with this PR, @serhiy-storchaka?

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

**Pull request merged**